DataBase/main.cpp

```diff
@@ -0,0 +1,200 @@
+#include <iostream>
+#include <vector>
+#include <map>
+#include <sstream>
+#include <iomanip>
+#include <regex>
+#include <set>
+
+class Date
+{
+public:
+    explicit Date(std::string&  date)
+    {
+        const std::regex r(R"(([+-]{0,1}(\d{1,}))-([+-]{0,1}(\d{1,}))-([+-]{0,1}(\d{1,})))");
+        if (!std::regex_match(date,r))
+        {
+            throw std::runtime_error("Wrong date format: " + date);
+        }
+        else
+        {
+            {
+                std::regex year(R"(([+-]{0,})\b(\d{1,}))");
+                std::smatch y;
+                std::regex_search(date, y, year);
+                m_year = std::stoi(y[0]);
+            }
+            {
+                std::regex month(R"(\b\-([+-]{0,})(\d{1,})\b)");
+                std::smatch m;
+                std::regex_search(date,m,month);
+                std::string stringMonth = m[0];
+                stringMonth.erase(0,1);
+                m_month = std::stoi(stringMonth);
+                if (m_month > 12 || m_month < 1)
+                    throw std::runtime_error("Month value is invalid: " + stringMonth);
+            }
+            {
+                std::regex day(R"(\b\-([+-]{0,})(\d{1,})$)");
+                std::smatch d;
+                std::regex_search(date,d,day);
+                std::string stringDay = d[0];
+                stringDay.erase(0, 1);
+                m_day = std::stoi(stringDay);
+                if (m_day > 31 || m_day < 1)
+                    throw std::runtime_error("Day value is invalid: " + stringDay);
+            }
+        }
+    }
+    int GetYear() const
+    {
+        return m_year;
+    };
```

Comment:
```suggestion
    }
```
Тут и ниже. После функций не надо точку с запятой. Компилятор хавает, потому что думает что это новая пустая строка, но по сути это мусор